### PR TITLE
Add depth `depthCompareFunction` for  Texture

### PR DESCRIPTION
**Pull request by GuoLei1990**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: packages/core/src/texture/Texture.ts [hunk 1]
```diff
@@ -11,10 +12,11 @@ import { TextureWrapMode } from "./enums/TextureWrapMode";
 export abstract class Texture extends RefObject {
   name: string;
 
-  /** @internal */
   _platformTexture: IPlatformTexture;
```
[the remark this answers is not in the document]
Comment (GuoLei1990) [reply packages/core/src/texture/Texture.ts:15]: done

File: packages/core/src/texture/enums/TextureFormat.ts [hunk 2]
```diff
@@ -59,8 +59,6 @@ export enum TextureFormat {
   /** Render to depth stencil buffer, engine will automatically select the supported precision. */
   DepthStencil,
   /** Render to stencil buffer. */
```
[the remark this answers is not in the document]
Comment (GuoLei1990) [reply packages/core/src/texture/enums/TextureFormat.ts]: done

File: packages/core/src/texture/RenderTarget.ts [hunk 3]
```diff
@@ -162,12 +162,21 @@ export class RenderTarget extends EngineObject {
     this._depth = depth;
 
     if (renderTexture) {
-      this._colorTextures = renderTexture instanceof Array ? renderTexture.slice() : [renderTexture];
+      const colorTextures = Array.isArray(renderTexture) ? renderTexture.slice() : [renderTexture];
```
Comment (zhuxudong) [inline packages/core/src/texture/RenderTarget.ts:165]: do not use `isArray` in engine

File: packages/core/src/texture/Texture2D.ts [hunk 4]
```diff
@@ -31,6 +31,15 @@ export class Texture2D extends Texture {
...
 
+    this._isDepthTexture =
+      format == TextureFormat.Depth ||
+      format == TextureFormat.DepthStencil ||
+      format == TextureFormat.Depth16 ||
+      format == TextureFormat.Depth24 ||
+      format == TextureFormat.Depth32 ||
+      format == TextureFormat.Depth24Stencil8 ||
+      format == TextureFormat.Depth32Stencil8;
+
```
Comment (zhuxudong) [inline packages/core/src/texture/Texture2D.ts:42]: can it use `this._isDepthTexture >= TextureFormat.Depth` ?

**Review by zhuxudong — APPROVED** [review]
+1

**Pull request merged**